src/base/bittorrent/tracker.cpp

```diff
@@ -240,7 +370,7 @@ void Tracker::replyWithPeerList(const TrackerAnnounceRequest &annonceReq)
     libtorrent::entry::list_type peerList;
     foreach (const Peer &p, peers) {
         //if (p != annonce_req.peer)
-        peerList.push_back(p.toEntry(annonceReq.noPeerId));
+        peerList.push_back(peerToEntry(p, annonceReq.noPeerId, annonceReq.peer.ip));
```

Comment:
>I am bit confused here. I tracked back where annonceReq.peer.ip comes from and I think it is the IP of the client requesting the peer list. So why does peerToEntry() potentially return his IP back?

Damnit. I messed up reading the code. Ignore The above. I'll come back if I figure it out.